core/src/main/java/org/ehcache/spi/cache/AbstractValueHolder.java

```diff
@@ -129,9 +133,20 @@ public boolean equals(Object obj) {
   }
 
   @Override
-  public float hitRate(TimeUnit unit) {
-    //XXX
-    return 0.0f;
+  public float hitRate(long now, TimeUnit unit) {
+    final long endTime = TimeUnit.NANOSECONDS.convert(now, TimeUnit.MILLISECONDS);
+    final long startTime = TimeUnit.NANOSECONDS.convert(creationTime, nativeTimeUnit());
+    float duration = (endTime - startTime)/(float)TimeUnit.NANOSECONDS.convert(1, unit);
+    return (hits/duration);
+  }
+
+  @Override
+  public long getHits() {
+    return this.hits;
+  }
+
+  public void setHits(long hits) {
+    this.hits = hits;
```

Comment:
This method should be protected and most likely use the updater instead of direct field access.